src/js/components/common/Header.js

```diff
@@ -18,54 +16,76 @@ class Header extends React.Component {
     document.removeEventListener('scroll', this.handleScroll);
   }
 
-  stickyToggle(sticky) {
-    const stickyHeight = sticky.clientHeight;
-    const stickyTop = sticky.offsetTop;
-    const scrollTop = document.body.scrollTop;
-
-    if (scrollTop > stickyHeight + stickyTop){
-      utils.addClass(sticky, 'is-sticky');
-    } else if (scrollTop <= stickyTop) {
-      utils.removeClass(sticky, 'is-sticky');
-    }
-  }
-
   navLinkOnClick(event) {
-    let href = event.target.getAttribute('href');
+    let target = $(event.target).attr('href') ? event.target : $(event.target).parents(".page-scroll")[0];
+    let href = $(target).attr('href');
     $('html, body').stop().animate({
         scrollTop: ($(href).offset().top - 50)
     }, 1250);
     event.preventDefault();
   }
 
+  isHomeLink() {
+    return window.location.pathname === '/';
+  }
+
+  generateDropdownLink(path, name, linkMap) {
+    let links = Object.keys(linkMap).map((key, index) => {
+      return (
+        <Link key={index} to={`#${key}`} className="page-scroll dropdown-item" onClick={this.navLinkOnClick}>{linkMap[key]}</Link>
+      )
+    });
+    if (window.location.pathname === path) {
+      return (
+        <li className="nav-item dropdown">
+          <Link className="nav-link dropdown-toggle" onlyActiveOnIndex activeClassName="active" to="#" data-href={path} id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            {name} <span className="caret down"></span>
+          </Link>
+          <div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
+            {links}
+          </div>
+        </li>
+      )
+    }
+    return (
+      <li className="nav-item">
+        <Link to={path} activeClassName="active" onlyActiveOnIndex className="nav-link">{name}</Link>
+      </li>
+    )
+  }
+
```

Comment:
Using JQuery within react, especially the global object makes testing a bit difficult as components are harder to mock.
Might `addClass` be a valid method on `event.target` (see line 64)?